src/DaiReserve.sol

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.7;
+
+import {ERC20Reserve, IERC20Reserve} from "./ERC20Reserve.sol";
+import {IDai} from "./Dai.sol";
+
+interface IDaiReserve is IERC20Reserve {
+    function dai() external view returns (IDai);
+}
+
+contract DaiReserve is ERC20Reserve, IDaiReserve {
```

Comment:
Why is it important to have a dedicated Dai reserve?

Reply:
Mostly for clarity and type safety, so we don't need to blindly cast `IERC20` to `IDai`. It's not very important, but it's not a lot of complex code either.